docs/beginners-guide-python.md

```diff
@@ -0,0 +1,219 @@
+</div>
+<div data-lang="Linux" markdown="1">
+ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Linuxbrew/install/master/install)"
+</div>
+</div>
+  
+  2. **Install Java (need Java 8).**
+ 
+<div class="codetabs">
+<div data-lang="OSX" markdown="1">
+{% highlight bash %}
+brew tap caskroom/cask
+brew install Caskroom/cask/java
+{% endhighlight %}
+</div>
+<div data-lang="Linux" markdown="1">
+brew tap caskroom/cask
+brew install Caskroom/cask/java
+</div>
+</div>
+
+  3. **Install Spark.**
+
+<div class="codetabs">
+<div data-lang="OSX" markdown="1">
+{% highlight bash %}
+brew install apache-spark
+{% endhighlight %}
+</div>
+<div data-lang="Linux" markdown="1">
+brew install apache-spark
+</div>
+</div>
+  
+
+  4. **Download SystemML.**
+
+<div class="codetabs">
+<div data-lang="OSX" markdown="1">
+{% highlight bash %}
+pip install SystemML
+curl "https://sparktc.ibmcloud.com/repo/latest/SystemML.jar" -o "SystemML.jar"
+{% endhighlight %}
+</div>
+<div data-lang="Linux" markdown="1">
+pip install SystemML
+wget https://sparktc.ibmcloud.com/repo/latest/SystemML.jar
+</div>
+</div>
+
+**Congrats! You can now use SystemML!**
+
+## Invoke SystemML's algorithms
+
+{% highlight python %}
+# Scikit-learn way
+from sklearn import datasets, neighbors
+from SystemML.mllearn import LogisticRegression
+from pyspark.sql import SQLContext
+sqlCtx = SQLContext(sc)
+digits = datasets.load_digits()
+X_digits = digits.data
+y_digits = digits.target + 1
+n_samples = len(X_digits)
+X_train = X_digits[:.9 * n_samples]
+y_train = y_digits[:.9 * n_samples]
+X_test = X_digits[.9 * n_samples:]
+y_test = y_digits[.9 * n_samples:]
+logistic = LogisticRegression(sqlCtx)
+print('LogisticRegression score: %f' % logistic.fit(X_train, y_train).score(X_test, y_test))
+{% endhighlight %}
+
+{% highlight python %}
+# MLPipeline way
+from pyspark.ml import Pipeline
+from SystemML.mllearn import LogisticRegression
+from pyspark.ml.feature import HashingTF, Tokenizer
+from pyspark.sql import SQLContext
+sqlCtx = SQLContext(sc)
+training = sqlCtx.createDataFrame([
+    (0L, "a b c d e spark", 1.0),
+    (1L, "b d", 2.0),
+    (2L, "spark f g h", 1.0),
+    (3L, "hadoop mapreduce", 2.0),
+    (4L, "b spark who", 1.0),
+    (5L, "g d a y", 2.0),
+    (6L, "spark fly", 1.0),
+    (7L, "was mapreduce", 2.0),
+    (8L, "e spark program", 1.0),
+    (9L, "a e c l", 2.0),
+    (10L, "spark compile", 1.0),
+    (11L, "hadoop software", 2.0)
+], ["id", "text", "label"])
+tokenizer = Tokenizer(inputCol="text", outputCol="words")
+hashingTF = HashingTF(inputCol="words", outputCol="features", numFeatures=20)
+lr = LogisticRegression(sqlCtx)
+pipeline = Pipeline(stages=[tokenizer, hashingTF, lr])
+model = pipeline.fit(training)
+test = sqlCtx.createDataFrame([
+    (12L, "spark i j k"),
+    (13L, "l m n"),
+    (14L, "mapreduce spark"),
+    (15L, "apache hadoop")], ["id", "text"])
+prediction = model.transform(test)
+prediction.show()
+{% endhighlight %}
+
+## Matrix operations
```

Comment:
Should matrix operations come above the description of machine learning algorithms?